dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -144,13 +209,75 @@ public void shouldSaveRuleAction() throws DotSecurityException, DotDataException
         final Host host = new SiteDataGen().nextPersisted();
         final Rule rule = new RuleDataGen().host(host).nextPersisted();
 
-        this.addPermission(role, host, true);
+        this.addRulesPermission(role, host, true);
+
+        final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
+
+        rulesAPI.saveRuleAction(ruleAction, user, false);
+
+        final List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        checkActions(ruleAction, allRules);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action in a Rule's Page
+     * Should: Save it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveRuleActionInPage() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+
+         final Template template = new TemplateDataGen().nextPersisted();
+         final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .nextPersisted();
+
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
 
+        this.addRulesPermission(role, host, true);
+        this.addPermission(role, htmlPageAsset, PermissionLevel.PUBLISH.getType());
         final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
 
         rulesAPI.saveRuleAction(ruleAction, user, false);
 
         final List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        checkActions(ruleAction, allRules);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action but without PUBLISH permission over the page
+     * Should: throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test (expected = DotSecurityException.class)
+    public void shouldSaveRuleActionInPageShouldNotWork() throws DotSecurityException, DotDataException {
```

Comment:
rename method to: `shouldNotSaveRuleActionInRuleOnPage`

Reply:
done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aR271